numpy/fft/_pocketfft_umath.c

```diff
@@ -235,15 +240,17 @@ irfft_loop(char **args, npy_intp const *dimensions, npy_intp const *steps, void
         double *op_double = (double *)op_or_buff;
         /* Copy complex input to buffer in FFTpack order */
         op_double[0] = ((double *)ip)[0];
-        if (nin > 1) {
+        if (npts > 1) {
             copy_data(ip + step_in, step_in, nin - 2,
                       (char *)&op_double[1], sizeof(npy_cdouble), npts / 2 - 1,
                       sizeof(npy_cdouble));
-            npy_intp ncopied = (npts / 2 - 1  < nin - 2 ? npts / 2 - 1 : nin - 2);
-            double *last = (double *)(ip + (ncopied + 1) * step_in);
-            op_double[ncopied * 2 + 1] = last[0];
-            if (npts % 2 == 1) {  /* For odd n, we still imag real of the last point */
-                op_double[ncopied * 2 + 2] = last[1];
+            npy_intp i_last = (npts / 2  < nin - 1 ? npts / 2 : nin - 1);
+            double *last = (double *)(ip + i_last * step_in);
+            if (i_last > 0) {
+                op_double[i_last * 2 - 1] = last[0];  /* real of the last point */
+            }
+            if (npts % 2 == 1) {  /* For odd n, we still imag of the last point */
```

Comment:
Comment is mangled slightly, but OK.

Reply:
I'll try to get it. And might as well add a comment about what FFTPACK format actually is, since it helps to understand what is done...

Reply:
I think of it in terms of the real transform taking advantage of the Hermitian symmetry of the spectrum to only include the lower half. The question is then how to reconstruct the full spectrum, which depends on whether the center of symmetry is at the highest frequency, or half a sample to the right. This same sort of question involving reflection symmetry turns up in many places.